cost/turbonomics/scale_virtual_machines_recommendations/aws/turbonomics_scale_virtual_machines.pt

```diff
@@ -0,0 +1,396 @@
+
+#get turbonomic recommendation data
+datasource "ds_get_turbonomics_recommendations" do
+  request do
+    run_script $js_get_turbonomics_recommendations, $auth_cookie
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "[*]") do
+      field "uuid", jmes_path(col_item, "uuid")
+      field "vendorIDValues", jmes_path(col_item, "target.vendorIds")
+      field "businessUUID", jmes_path(col_item, "target.aspects.cloudAspect.businessAccount.uuid")
+      field "resourceGroup", jmes_path(col_item, "target.aspects.cloudAspect.resourceGroup.displayName")
+      field "resourceName", jmes_path(col_item, "target.displayName")
+      field "region", jmes_path(col_item, "currentLocation.displayName")
+      field "resourceType", jmes_path(col_item, "currentEntity.displayName")
+      field "newResourceType", jmes_path(col_item, "newEntity.displayName")
+      field "provider", jmes_path(col_item, "target.discoveredBy.type")
+      field "details", jmes_path(col_item, "details")
+      field "tags", jmes_path(col_item, "target.tags")
+      field "createdTime", jmes_path(col_item, "createTime")
+      field "savings", jmes_path(col_item, "stats[0].value")
+      field "savingsCurrency", jmes_path(col_item, "stats[0].units")
+    end
+  end
+end
+
+##this will potentially be a lot of calls. how to make this more performant
+datasource "ds_get_action_details" do
+  request do
+    run_script $js_get_action_details, $auth_cookie, $ds_get_turbonomics_recommendations
+  end
+  result do
+    encoding "json"
+    field "uuids", response
+  end
+end
+
+##this will potentially be a lot of calls. how to make this more performant
+datasource "ds_get_business_units" do
+  request do
+    run_script $js_get_business_units, $auth_cookie, $param_provider
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "[*]") do
+      field "uuid", jmes_path(col_item, "uuid")
+      field "accountID", jmes_path(col_item, "accountId")
+      field "displayName", jmes_path(col_item, "displayName")
+    end
+  end
+end
+
+datasource "ds_filtered_turbonomics_recommendations" do
+  run_script $js_filtered_turbonomics_recommendations, $ds_get_turbonomics_recommendations, $ds_get_action_details, $ds_get_business_units, $param_provider
+end
+
+###############################################################################
+# Scripts
+###############################################################################
+
+##script using a manually acquired turbonomic cookie and pulls hard coded scalable vm data
+script "js_get_turbonomics_recommendations", type: "javascript" do
+  result "request"
+  parameters "auth_cookie"
+  code <<-EOS
+  //replace cookie every day this is run
+  //up this limit value to 1000 etc when done with POC testing
+    var request = {
+      verb: "POST",
+      pagination: "turbonomics_vm_pagination_header",
+      host: "sales1.demo.turbonomic.com",
```

Comment:
Get this value from a parameter, otherwise we force user use "sales1.demo.turbonomic.com" always.

Reply:
Done. That host field is now a parameter.